### PR TITLE
This deploys and configures metallb on ocp-prod

### PR DESCRIPTION
**Pull request by naved001**
It's mostly based on https://github.com/operate-first/apps/pull/405
and https://gitlab.com/naved001/moc-cnv-sandbox/-/tree/moc-ocp/roles/ocp/metallb

The secret handling is different, because we are using ExternalSecrets controllers
in our case.

### CODE REVIEW

**Comment by larsks** [comment]
I'd like to take a look at this before it merges, but that probably won't be until Monday :)

**Comment by naved001** [comment]
I am marking this as draft then.

File: cluster-scope/overlays/ocp-prod/externalsecrets/metallb-memberlist-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+apiVersion: "kubernetes-client.io/v1"
+kind: ExternalSecret
+metadata:
+  name: memberlist
+  namespace: metallb-system
+spec:
+  backendType: secretsManager
+  data:
+    - key: cluster/ocp-prod/metallb/memberlist
+      name: secretkey
```
Comment (larsks) [inline cluster-scope/overlays/ocp-prod/externalsecrets/metallb-memberlist-secret.yaml:10]: See my comments on #35 regarding the structure of secrets in AWS.

Comment (naved001) [reply cluster-scope/overlays/ocp-prod/externalsecrets/metallb-memberlist-secret.yaml:10]: I fixed this too.

**Comment by naved001** [comment]
I am just questioning myself. Why aren't we installing this with helm? it looks like it can take care of creating most resources.

https://metallb.universe.tf/installation/#installation-with-helm

**Comment by larsks** [comment]
> Why aren't we installing this with helm?

Because we really want this repository to be the source of truth for what we deploy in OpenShift. I mean, they provide instructions for deploying with kustomize as well (https://metallb.universe.tf/installation/#installation-with-helm) referring directly to the remote resources; we could absolutely do that, but the concern is the same as, say, loading a javascript library from a remote source: what if the remote source goes down? Changes unexpectedly? Etc.

We can chat about how strict we want to be. Maybe during our 11am today?

**Comment by larsks** [comment]
Based on our discussion, we can try going the "include from github route", as documented at that installation-with-kustomize link. [This post](https://blog.oddbit.com/post/2020-09-27-installing-metallb-on-openshif/) may be of interest w/r/t necessary patches.

**Pull request closed without merging**